lightningd/peer_htlcs.c

```diff
@@ -779,14 +779,28 @@ htlc_accepted_hook_callback(struct htlc_accepted_hook_payload *request,
 	enum htlc_accepted_result result;
 	enum onion_type failure_code;
 	u8 *channel_update;
+	bool valid;
 	result = htlc_accepted_hook_deserialize(buffer, toks, &payment_preimage, &failure_code, &channel_update);
 
+	/* In order to handle it internally the payload needs to have at least
+	 * the `outgoing_cltv_value`, the `amt_to_forward` and the
+	 * `short_channel_id` if we are forwarding. */
```

Comment:
Please quote the BOLT directly instead, so we know when this requirement changes (which it will!).